kedro/io/data_catalog.py

```diff
@@ -170,8 +184,12 @@ def __init__(
         self._data_sets = dict(data_sets or {})
         self.datasets = _FrozenDatasets(self._data_sets)
         self.layers = layers
+        # Keep a record of all patterns in the catalog.
+        # {dataset pattern name : dataset pattern body}
+        self._dataset_patterns = dict(dataset_patterns or {})
```

Comment:
This still confuse me a lot, is there any difference? Isn't dataset_patterns a dict already?
```suggestion
        self._dataset_patterns =dataset_patterns or {}
```